torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -305,7 +305,7 @@ class ExprSegmentationSorter {
 
   std::deque<ExprGroup*> to_visit_;
 
-  std::unordered_set<ExprGroup*> to_merge_;
+  std::set<ExprGroup*> to_merge_;
```

Comment:
I assume the ordering is determined by memory addresses, so it's not run-to-run deterministic.

We can use `VectorOfUniqueEntries` in disjoint_set.h.

Reply:
Yes it is about ordering in the data structure. `std::set` is the closest to the original use we have, I think. I assume `VectorOfUniqueEntries` doesn't support random access insertion/deletion, so would be harder to use here.

Reply:
I guess we still don't want to order pointers by default. Doesn't sound write. Maybe comparing the names would help, assuming everything else is deterministic : ).

Reply:
It doesn't, but `std::set` isn't guaranteed to be deterministic. I think determinism in expr sorting is really important, so we should do something else. Perhaps, defining some deterministic ordering between `ExprGroup*`?

Reply:
Maybe set a unique integer number to each instance of `ExprGroup`?  It seems `ExprGroup` is always created by `ExprSegmentationSorter`, so we could pass an increasing number of integer to the `ExprGroup` constructor.

Reply:
Was thinking about that but later realized the use cases of `to_merge_` didn't require random access at all. So just switched to the way `FusionSegmenter` did it.